test/unit/auth/auth.spec.ts

```diff
@@ -1915,6 +1964,839 @@ describe('Auth', () => {
+            // Confirm expected error returned.
+            expect(error).to.equal(expectedError);
+          });
+      });
+    });
+  });
+
+  describe('listProviderConfigs()', () => {
+    const options: AuthProviderConfigFilter = {
+      type: 'oidc',
+    };
+    let stubs: sinon.SinonStub[] = [];
+
+    afterEach(() => {
+      _.forEach(stubs, (stub) => stub.restore());
+      stubs = [];
+    });
+
+    it('should be rejected given no options', () => {
+      return (auth as any).listProviderConfigs()
+        .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
+    });
+
+    it('should be rejected given an invalid AuthProviderConfigFilter type', () => {
+      const invalidOptions = {
+        type: 'unsupported',
+      };
+      return (auth as Auth).listProviderConfigs(invalidOptions as any)
+        .then(() => {
+          throw new Error('Unexpected success');
+        })
+        .catch((error) => {
+          expect(error).to.have.property('code', 'auth/argument-error');
+        });
+    });
+
+    it('should be rejected given an app which returns null access tokens', () => {
+      return (nullAccessTokenAuth as Auth).listProviderConfigs(options)
+        .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+    });
+
+    it('should be rejected given an app which returns invalid access tokens', () => {
+      return (malformedAccessTokenAuth as Auth).listProviderConfigs(options)
+        .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+    });
+
+    it('should be rejected given an app which fails to generate access tokens', () => {
+      return (rejectedPromiseAccessTokenAuth as Auth).listProviderConfigs(options)
+        .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+    });
+
+    describe('using OIDC type filter', () => {
+      const expectedError = new FirebaseAuthError(AuthClientErrorCode.INTERNAL_ERROR);
+      const pageToken = 'PAGE_TOKEN';
+      const maxResults = 50;
+      const filterOptions: AuthProviderConfigFilter = {
+        type: 'oidc',
+        pageToken,
+        maxResults,
+      };
+      const listConfigsResponse: any = {
+        oauthIdpConfigs : [
+          getOIDCConfigServerResponse('oidc.provider1'),
+          getOIDCConfigServerResponse('oidc.provider2'),
+        ],
+        nextPageToken: 'NEXT_PAGE_TOKEN',
+      };
+      const expectedResult: any = {
+        providerConfigs: [
+          new OIDCConfig(listConfigsResponse.oauthIdpConfigs[0]),
+          new OIDCConfig(listConfigsResponse.oauthIdpConfigs[1]),
+        ],
+        pageToken: 'NEXT_PAGE_TOKEN',
+      };
+      const emptyListConfigsResponse: any = {
+        oauthIdpConfigs: [],
+      };
+      const emptyExpectedResult: any = {
+        providerConfigs: [],
+      };
+
+      it('should resolve on listOAuthIdpConfigs request success with configs in response', () => {
```

Comment:
I'd rephrase this as `should resolve on success with configs in response`. Similar changes elsewhere.

Basically, `listOAuthIdpConfigs` is an internal implementation detail that we have to mock out, and there's no need to mention it everywhere.